grails-app/services/au/org/ala/biocollect/merit/DocumentService.groovy

```diff
@@ -100,4 +102,65 @@ class DocumentService {
 
         documents;
     }
+
+    /**
+     * Returns true if the current user has permission to edit/update the
+     * supplied document.
+     * @param document the document to be edited/updated
+     */
+    boolean canEdit(Map document) {
+        if (document.documentId) {
+            document = get(document.documentId)
+        }
+        boolean canEdit = false
+
+        if (document) {
+            // Only FC_ADMINS can edit an existing read only document, but
+            // other roles can create them.
+            if (document.readOnly && document.documentId) {
+                canEdit = userService.userIsAlaOrFcAdmin()
+            }
+            else {
+                canEdit = hasEditorPermission(document)
+            }
+
+        }
+
+        canEdit
+    }
+
+    /** Returns true if the currently logged in user has permission to edit the supplied Document */
+    private boolean hasEditorPermission(Map document) {
+        boolean canEdit = userService.userIsAlaOrFcAdmin()
+        if (!canEdit) {
+            // Check the permissions that apply to the entity the document is
+            // associated with.
+            String userId = userService.getCurrentUserId()
+            if (document.projectId) {
+                canEdit = userService.canUserEditProject(userId, document.projectId)
+            } else if (document.organisationId) {
+                canEdit = userService.isUserAdminForOrganisation(userId, document.organisationId)
+            }
+            else if (document.activityId) {
+                Map activity = activityService.get(document.activityId)
+                if (activity?.projectId) {
+                    canEdit = userService.canUserEditProject(userId, activity.projectId)
+                }
+            }
+        }
+        canEdit
+    }
+    /** Returns true if the currently logged in user has permission to view the supplied Document */
+    boolean canView(Map document) {
+        document.publiclyViewable || userService.userHasReadOnlyAccess() || hasEditorPermission(document)
```

Comment:
Will this satisfy the condition a sighting photo uploaded is viewable by everyone (logged in and not logged in users)? I am not certain publiclyViewable flag will be set. Another use case is logo and banner of a project should be viewable by everyone.

Reply:
How can the system differentiate between a photo attached to a sighting that is supposed to be public vs a document attached to a works activity that is not?  Also, how should embargoed records be handled?  I propose that since this is an improvement to the current situation I can make documents with an activityId publicly viewable for now and it can be addressed when the other document changes are made.

Reply:
Banners etc, should have the publiclyViewable flag set in ecodata before returning the document to biocollect.

Reply:
That sounds good.